src/dev/encoder.cpp

```diff
@@ -0,0 +1,152 @@
+#include <EVT/dev/encoder.hpp>
+
+namespace EVT::core::DEV {
+
+Encoder::Encoder(IO::GPIO& a, IO::GPIO& b, int64_t range, int64_t initialPosition) : a(a), b(b), range(range), position(initialPosition) {
+    //making sure range is positive
+    if (range < 0) {
```

Comment:
This type of error handling is going to be really hard to debug. If you end up restructuring the class the way I described in another comment, this will already get changed, but if not, definitely find another way of handling this